lighthouse-cli/cli-flags.js

```diff
@@ -141,14 +142,18 @@ function getFlags(manualArgv) {
         // Lighthouse doesn't need a URL if...
         //   - We're in auditMode (and we have artifacts already)
         //   - We're just listing the available options.
+        //   - We're just printing the config.
         // If one of these don't apply, stop the program and ask for a url.
         const isListMode = argv.listAllAudits || argv.listTraceCategories;
         const isOnlyAuditMode = !!argv.auditMode && !argv.gatherMode;
-        if (!isListMode && !isOnlyAuditMode && argv._.length === 0) {
-          throw new Error('Please provide a url');
+        const isPrintConfigMode = argv.printConfig;
+        if (isListMode || isOnlyAuditMode || isPrintConfigMode) {
```

Comment:
i think printConfigMode is actually captured under the spirit of the `isListMode` variable. should be renamed though.   `doPrintAndQuit` ?

Reply:
done